src/model_signing/signing/signing.py

```diff
@@ -55,98 +56,85 @@
     from typing_extensions import Self
 
 
-def _convert_descriptors_to_direct_statement(
-    model_manifest: manifest.Manifest, predicate_type: str
-) -> statement.Statement:
-    """Converts manifest descriptors to an in-toto statement, as subjects.
-
-    Args:
-        manifest: The manifest to extract the descriptors from. Assumed valid.
-        predicate_type: The predicate_type to use in the in-toto statement.
-    """
-    subjects = []
-    for descriptor in model_manifest.resource_descriptors():
-        subject = statement.ResourceDescriptor(
-            name=descriptor.identifier,
-            digest={"sha256": descriptor.digest.digest_hex},
-            annotations={"actual_hash_algorithm": descriptor.digest.algorithm},
-        )
-        subjects.append(subject.pb)
-
-    return statement.Statement(
-        subjects=subjects,
-        predicate_type=predicate_type,
-        # https://github.com/in-toto/attestation/issues/374
-        predicate={"unused": "Unused, just passed due to API requirements"},
-    )
-
-
 class SigningPayload:
-    """In-toto payload where the subjects are the model files themselves.
-
-    This payload is supposed to be used for manifests where every file in the
-    model is matched with a digest. Because existing tooling only supports
-    established hashing algorithms, we annotate every subject with the actual
-    hash algorithm used to compute the file digest, and use "sha256" as the
-    algorithm name in the digest itself.
-
-    Example (TODO: needs to be updated):
+    """In-toto payload used to represent a model for signing.
+
+    This payload represents all the object (files, shards, etc.) of the model
+    paired with their hases. It can be seen as a serialization of a manifest.
+    The hashes are all recorded under the predicate, given that for the subject
+    we are limited on what hashes we can use
+    (https://github.com/sigstore/sigstore-python/issues/1018). Each hash follows
+    the format of a ResourceDescriptor: is an object containing a name for the
+    object, the hashing algorithm, and the digest value. These are recorded in
+    the predicate, as part of the `"resources"` list.
+
+    The subject is a name for the model (taken from the model's directory) and a
+    global digest over all the computed digests. This is SHA256 computed over
+    all the digests, in the order they show up in the predicate (we canonicalize
+    this to be in alphabetical order). This digest can be used to refer to the
```

Comment:
note to self: just based on digests? or also the file header like what digestofdigests was?

Reply:
and merge hasher is no longer customizable, always sha256?

Reply:
Just the digests, since at this point the names are already recorded below so we catch models where the name changes.

This has to be a hash that sigstore accepts (https://github.com/sigstore/sigstore-python/issues/1018), so always using sha256 is better than allowing customization.

This is somewhat what we used to have as `DigestOfDigests` payload